In [56]:
import pandas as pd

In [57]:
pair_wise_in = pd.read_csv("../datasets/Pairwise_HIT.csv") 
pair_wise_out1 = pd.read_csv("../datasets/Pairwise Batch 2.csv")
pair_wise_out2 = pd.read_csv("../datasets/Pairwise Batch 1 50.csv")

In [58]:
pair_wise_in

Unnamed: 0                                            Record1  \
0            0  Case Logic 15.6-Inch Laptop and iPad Slim Case...   
1            1  Royce Leather Unisex Pilot / Catalog Computer ...   
2            2  Acer Aspire V 15 Touch V3-572P-326T Signature ...   
3            3  Kensington K64663US ClickSafe Master Keyed Lap...   
4            4  Lenovo G410 Laptop Computer - 59410765 - Black...   
..         ...                                                ...   
95          95  Case Logic VLS-114 13-Inch, 14-Inch Laptop Sle...   
96          96  Case Logic 15.6-Inch Laptop Backpack (Anthraci...   
97          97  Case Logic Sleeve for 15.6-Inch Notebook, Purp...   
98          98  17 Inch Red Flowers Floral Design Laptop Compu...   
99          99  Kingston HyperX Impact Black 8GB Kit (2x4GB) 2...   

                                              Record2  label  
0   Case Logic TopLoading Laptop Case Black VNAI-2...      1  
1                                 Royce Leather 6375       1  
2   Acer 15.6 Aspire Notebook 6 GB Memory 1 TB Har...      0  
3                              Agptek.com, Inc LR64F       0  
4   Lenovo 14 Laptop Intel Core i7 8GB Memory 1TB+...      1  
..                                                ...    ...  
95          Case Logic Laptop Sleeve Pink LAPS-116PI       1  
96        Case Logic Laptop Backpack Gray DLBP-116DA       1  
97  Case Logic 17 Laptop Sleeve Black Black LAPS-117       1  
98      Lenmar LBZ352HP Notebook Batt Black LBZ352HP       0  
99                  Cocoon Innovations, LLC CPS380GY       0  

[100 rows x 4 columns]

In [59]:
pair_wise_out=pd.concat([pair_wise_out1,pair_wise_out2])

# filtered = pair_wise_out[pair_wise_out["Input.Record1"] == "Patriot Signature 4 GB PC2-6400 DDR2 Notebook Memory PSD24G8002S"]
# filtered[["WorkerId", "Input.Record1", "Input.Record2", "Input.label", "Answer.equal.label"]]


# pair_wise_out = pair_wise_out[pair_wise_out["WorkerId"] != "A373U4XYXC178J"]
# pair_wise_out = pair_wise_out[pair_wise_out["WorkerId"] != "A3FFYWZLVBQDM6"]
# pair_wise_out = pair_wise_out[pair_wise_out["WorkerId"] != "A2ZCIU7Z17MHHZ"]


# Patriot Signature 4 GB PC2-6400 DDR2 Notebook Memory PSD24G8002S
# Patriot Memory Patriot Signature DDR2 4GB CL PSD24G8002S
# They are the same

In [60]:
rows=list(pair_wise_out["Input.Record1"].value_counts()[:25].index)

In [61]:
labels=[]
for i in rows:
    df = pair_wise_out.loc[pair_wise_out["Input.Record1"]==i]
    labels.append(df["Answer.equal.label"].values[:7])

In [62]:
true_labels=[]
for i in rows:
    df= pair_wise_in.loc[pair_wise_in["Record1"]==i]
    true_labels.append(df.label.values[0])

In [63]:
true_labels

[1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1]

In [64]:
worker_labels=[]
for i in labels:
    numeric_ol = []
    for j in i:
        if j=="Yes":
            numeric_ol.append(1)
        else:
            numeric_ol.append(0)
    worker_labels.append(numeric_ol)        

In [65]:
final_worker_labels=[]

for i in worker_labels:
    
    final_worker_labels.append(pd.Series(i).value_counts().index[0])

In [66]:
import numpy as np
from sklearn.metrics import confusion_matrix

In [67]:
tn, fp, fn, tp = confusion_matrix(true_labels, final_worker_labels).ravel()

In [68]:
tn, fp, fn, tp

(11, 0, 13, 1)

In [69]:
Precision=tp/(tp+fp)
Precision

1.0

In [71]:
Recall=tp/(tp+fn)
Recall

0.07142857142857142

In [72]:
f_score=(2 * Precision * Recall) / (Precision + Recall)
f_score

0.13333333333333333

# With Honeypot:
#### Recall = 0.142857
#### Precision = 1
#### f_score = 0.25

# Without Honeypot:
#### Recall = 0.07142857
#### Precision = 1
#### f_score = 0.13333